### PR TITLE
Invoke retrieval lambda from UI

### PR DESCRIPTION
**Pull request by attwad**
For #781 

Also removing the docker config printout statement because it prints secret keys to the terminal, not great.

Not quite sure yet how to handle that nicely in the UI, for now it just prints to the console the bucket and key.

### CODE REVIEW

File: verification/curator-service/ui/src/components/SourceTable.test.tsx [hunk 1]
```diff
@@ -5,10 +5,18 @@ import { fireEvent, render } from '@testing-library/react';
 import React from 'react';
 import SourceTable from './SourceTable';
 import axios from 'axios';
+import User from './User';
 
 jest.mock('axios');
 const mockedAxios = axios as jest.Mocked<typeof axios>;
 
+const curator: User = {
```
Comment (axmb) [inline verification/curator-service/ui/src/components/SourceTable.test.tsx:13]: Worth adding a unit test for the retrieval feature? Or a case in the cypress spec.

File: verification/curator-service/api/src/clients/aws-lambda-client.ts [hunk 2]
```diff
@@ -65,4 +73,31 @@ export default class AwsLambdaClient {
             .removePermission(removePermissionParams)
             .promise();
     };
+
+    /**
+     * Invoke retrieval function lambda synchrnously, returning its output. */
```
Comment (axmb) [inline verification/curator-service/api/src/clients/aws-lambda-client.ts:78]: nit, sp

File: verification/curator-service/api/test/clients/aws-lambda-client.test.ts [hunk 3]
```diff
@@ -81,3 +86,33 @@ describe('removePermission', () => {
...
+        expect(res).toEqual(payload);
+    });
+    it('throws when error is returned by aws api', async () => {
+        const expectedError = new Error('AWS error');
+        invokeSpy.mockRejectedValueOnce(expectedError);
+        return expect(client.invokeRetrieval('some-source-id')).rejects.toThrow(
+            expectedError,
+        );
+    });
+    it('throws when error zhen function is not run properly', async () => {
```
Comment (axmb) [inline verification/curator-service/api/test/clients/aws-lambda-client.test.ts:110]: nit, sp

File: verification/curator-service/api/openapi/openapi.yaml [hunk 4]
```diff
@@ -167,6 +167,32 @@ paths:
...
+        "200":
+          $ref: "#/components/responses/200Retrieval"
+        "400":
+          $ref: "#/components/responses/400"
+        "403":
+          $ref: "#/components/responses/403"
+        "404":
+          $ref: "#/components/responses/404"
+        "422":
+          $ref: "#/components/responses/422"
```
Comment (axmb) [inline verification/curator-service/api/openapi/openapi.yaml:193]: This doesn't throw a 422, I don't think.

File: verification/curator-service/api/src/controllers/sources.ts [hunk 5]
```diff
@@ -209,4 +211,17 @@ export default class SourcesController {
         res.status(204).end();
         return;
     };
+
+    /** Trigger retrieval of the source's content in S3. */
+    retrieve = async (req: Request, res: Response): Promise<void> => {
```
Comment (axmb) [inline verification/curator-service/api/src/controllers/sources.ts:216]: Worth adding a quick unit test for this?

File: verification/curator-service/ui/src/components/SourceTable.tsx [hunk 6]
```diff
@@ -274,6 +278,24 @@ class SourceTable extends React.Component<Props, SourceTableState> {
         return field?.trim() !== '';
     }
 
+    triggerRetrieval(id: string): Promise<unknown> {
+        return new Promise((resolve, reject) => {
+            const url = this.state.url + id + '/retrieve';
+            this.setState({ error: '' });
```
[the remark this answers is not in the document]
Comment (attwad) [reply verification/curator-service/ui/src/components/SourceTable.tsx:284]: ok, let me revert the UI changes for now though, I thought of something better that I'll work on today, just keeping the api in this PR.

File: verification/curator-service/ui/src/components/SourceTable.tsx [hunk 7]
```diff
@@ -549,6 +575,34 @@ class SourceTable extends React.Component<Props, SourceTableState> {
...
+                                              <span aria-label="retrieve">
+                                                  <CloudDownloadIcon />
+                                              </span>
+                                          ),
+                                          tooltip: 'Trigger retrieval',
+                                          onClick: (
+                                              _: any,
+                                              rows: any,
+                                          ): void => {
+                                              const promises = rows.map(
```
[the remark this answers is not in the document]
Comment (attwad) [reply verification/curator-service/ui/src/components/SourceTable.tsx:592]: yeah that made me think that a per-row action might be better, it's easier to display the result as well, sending you this in a separate PR.

File: verification/curator-service/ui/src/components/SourceTable.tsx [hunk 8]
```diff
@@ -549,6 +575,34 @@ class SourceTable extends React.Component<Props, SourceTableState> {
...
+                                                  (row: TableRow) =>
+                                                      this.triggerRetrieval(
+                                                          row._id,
+                                                      ),
+                                              );
+                                              Promise.all(promises).then(() => {
```
[the remark this answers is not in the document]
Comment (attwad) [reply verification/curator-service/ui/src/components/SourceTable.tsx:598]: this was to reest the selection but yeah we can live without it

File: verification/curator-service/ui/src/components/SourceTable.test.tsx (hunk 1 above)
Comment (attwad) [reply verification/curator-service/ui/src/components/SourceTable.test.tsx:13]: yup, doing that in the PR with the UI stuff in it next

File: verification/curator-service/api/test/clients/aws-lambda-client.test.ts (hunk 3 above)
Comment (attwad) [reply verification/curator-service/api/test/clients/aws-lambda-client.test.ts:110]: that new keyboard layout is killing me haha

File: verification/curator-service/api/openapi/openapi.yaml (hunk 4 above)
Comment (attwad) [reply verification/curator-service/api/openapi/openapi.yaml:193]: correct, removed.

File: verification/curator-service/api/src/controllers/sources.ts (hunk 5 above)
Comment (attwad) [reply verification/curator-service/api/src/controllers/sources.ts:216]: always worth it, done

**Pull request merged**